Lib/test/test_logging.py

```diff
@@ -4347,6 +4347,18 @@ def test_queue_listener(self):
         self.assertTrue(handler.matches(levelno=logging.CRITICAL, message='6'))
         handler.close()
 
+    @unittest.skipUnless(hasattr(logging.handlers, 'QueueListener'),
+                         'logging.handlers.QueueListener required for this test')
+    def test_queue_listener_context_manager(self):
+        handler = TestHandler(support.Matcher())
+        with logging.handlers.QueueListener(self.queue, handler) as listener:
+            self.assertIsNotNone(listener._thread)
```

Comment:
```suggestion
        with logging.handlers.QueueListener(self.queue, handler) as listener:
            self.assertIsIsintance(listener, logging.handlers.QueueListener)
            self.assertIsNotNone(listener._thread)
```

Just for checking self-return.